### ISSUE TITLE
Using signals for actions other that shutdown

### ISSUE THREAD

**Issue by humnandy** [swift-server/swift-service-lifecycle#64]
Is there support for trapping signals for anything other than shutdown hooks?

Linux services will often trap SIGHUP to reload configuration without stopping the service.

I believe that a Windows service can handle a SERVICE_ACCEPT_PARAMCHANGE state request to do the same thing.

This would be a useful feature for production services.

**Comment by ktoso** [comment]
This sounds like a reasonable addition to me, I guess you're after similar semantics like httpd or similar, i.e. 

> Sending the HUP or restart signal to the parent causes it to kill off its children like in TERM, but the parent doesn't exit. It re-reads its configuration files, and re-opens any log files. Then it spawns a new set of children and continues serving hits.

I can see that be useful indeed, esp. if you're managing some child processes...

Have you thought about how to surface it in API?
The current hooks are just a configured `var shutdownSignal: [Signal]?` though this would be more like an `lifecycle.registerSignalHandler(.HUP) { ... }` though it should likely not allow registering the ones used for shutdown etc...

WDYT?

Would you like to send in a PR for this?

**Comment by tomerd** [comment]
+1 on @ktoso design proposal

**Comment by hassila** [comment]
This would be nice, missed this issue when opening https://github.com/swift-server/swift-service-lifecycle/issues/81

**Comment by tomerd** [comment]
@hassila @humnandy `ServiceLifecycle` exposes a static method `trap` which can be used to setup signals handlers. Does the help with the need here?

**Comment by hassila** [comment]
@tomerd I think that would work perfectly fine, I just did a simple test and it works fine with a small change, see PR: https://github.com/swift-server/swift-service-lifecycle/pull/86
Should probably add SIGHUP too though.

**Comment by hassila** [comment]
I added support for HUP too.
So for reference if someone stumbles here, a trivial handler for HUP is done thusly:
```
        let alrmDispatchSourceSignal3 = ServiceLifecycle.trap(signal: ServiceLifecycle.Signal.HUP, handler: { signal in
            print("signal \(signal)")
        }, cancelAfterTrap: false)

```

**Comment by tomerd** [comment]
nice! should we close this issue once your PR lands?

**Comment by hassila** [comment]
@tomerd definitely, in case any additional signals would be needed in the future they are trivial to add. I think HUP and USR1/USR2 are the most useful ones.

**Comment by tomerd** [comment]
closed via #86
